aggregator/src/test/scala/weco/concepts/aggregator/ClusterConfWithSecretsTest.scala

```diff
@@ -0,0 +1,79 @@
+package weco.concepts.aggregator
+
+import org.scalatest.GivenWhenThen
+import org.scalatest.featurespec.AnyFeatureSpec
+import org.scalatest.matchers.should.Matchers
+import weco.concepts.aggregator.secrets.ClusterConfWithSecrets
+import weco.concepts.common.elasticsearch.ElasticAkkaHttpClient.ClusterConfig
+
+class ClusterConfWithSecretsTest
+    extends AnyFeatureSpec
+    with Matchers
+    with GivenWhenThen {
+
+  Feature("Resolving secret keys in ClusterConf") {
+    def resolver(keys: Seq[String]): Map[String, String] = {
+      keys.map(secretName => secretName -> secretName.reverse).toMap
+    }
+
+    val replacer = new ClusterConfWithSecrets(resolver)
+
+    Scenario("Resolving secrets") {
+      Given("a ClusterConfig instance with secrets")
+      val config = ClusterConfig(
+        scheme = "gopher",
+        host = "moc.elpmaxe",
+        port = 1234,
+        username = Some("Henry"),
+        password = Some("drowssap"),
+        resolveSecrets = true
+      )
+      When("ClusterConfWithSecrets is applied")
+      val resolvedConfig = replacer(config)
+      Then(
+        "the host and password secrets are resolved, leaving other value as they were"
+      )
+      resolvedConfig shouldBe ClusterConfig(
+        scheme = "gopher",
+        host = "example.com",
+        port = 1234,
+        username = Some("Henry"),
+        password = Some("password")
+      )
```

Comment:
minor: I would use `copy` on the case class to emphasise which fields have changed, i.e.

```suggestion
      resolvedConfig shouldBe config.copy(
        host = "example.com",
        password = Some("password")
      )
```

Reply:
Good thought.  The reason I wrote it out in full is because it uses config.copy in the SUT.  Were I being a perfectionist here, I'd actually inspect each property.